### ISSUE TITLE
Fail to build nextjs app since 20240407.1.0

### ISSUE THREAD

**Issue by tyx** [actions/runner-images#9700]
### Description

Hi,

I just don't get what is going on since a week and the new runner image 20240407.1.0 with ubuntu-22.04 but I am no longer able to build my nextjs app... 

Error looks around the locate detection : 
![image](https://github.com/actions/runner-images/assets/245494/1dbbf28c-ece5-4b85-8a94-a68f1d4a270e)

Please note we add no error with runner image `20240324.2.0` and the same build that pass with this image, no longer build with the new image. No change on our side. (I rerun an old build to be sure the problem was the runner image)

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20240407.1.0

### Is it regression?

20240324.2.0

### Expected behavior

I expect my nextjs app build

### Actual behavior

![image](https://github.com/actions/runner-images/assets/245494/1dbbf28c-ece5-4b85-8a94-a68f1d4a270e)

### Repro steps

I will try to isolate the issue but I have no idea for now what is going on and how an update of runner image could break a nextjs build.

**Comment by tyx** [comment]
Unfortunately it is a private repository. I am working on a repro step right now

**Comment by tyx** [comment]
After hours isolating the problem I was able to confirm that the problem was linked to the tricks in the nextjs doc to redirect our default locale to /en.

So we end up with a default locale in our `getStaticProps`, which up until now hasn't caused any problems because the result of the locale negotiation is that we end up with the default locale.

It seems that the problem is linked to `formatjs` and `i18n` and the way of determining whether a locale is valid or not.

I imagine it must be based on a system call and that's why we got the error on the new image and on a new docker image.

Sorry for the inconvenience!
